ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch new 25 results for fetch news.
+  function fetchNewNews() {
+    $topiclist.empty();
+    $.get("https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty")
+    .then(function(res){
+      hackerNews = hackerNews.concat(res.slice(0,25));
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch favorites
+
+  function fetchFavorites() {
+    $topiclist.empty();
+    $.ajax({
+          type: "Get",
+          url: "https://hn-favorites.herokuapp.com/stories.json",
+          headers: {
+            authorization: JSON.parse(localStorage.getItem("Nacker"))[0]
+          }
+        }).then(function(res) {
+          console.log(res);
+          hackerNews = hackerNews.concat(res);
+          hackerNews.forEach(function(value) {
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star",
+          });
+          $star.data("story_id",value.story_id);
+          $star.data("favorite_id",value.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: value.url,
+            text: value.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${value.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          });
+          hackerNews = [];
+
+        })
+  }
+
+  //Slide down forms
+  $login.on("click", function(event) {
+    $(".form-row").slideToggle(300);
+  })
+
+
+  //Top stories click
+  $topStories.on("click", function(event) {
+    fetchTopNews();
```

Comment:
In this case, you could just pass in `fetchTopNews` directly, rather than creating a new function to wrap `fetchTopNews`:

`$topStories.on("click", fetchTopNews);`

If you refactor those three functions into one that takes an argument, you can still use this pattern, though you may need to bind to pass in the arguments.